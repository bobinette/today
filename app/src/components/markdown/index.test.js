import React from 'react';
import { fromJS } from 'immutable';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { Markdown } from '.';

configure({ adapter: new Adapter() });

test('render the log list', () => {
  const fetchLogs = jest.fn();
  const props = {
    text: `# Title

    A little bit of text
    `,
  };

  const tree = shallow(<Markdown {...props} />);
  expect(tree).toMatchSnapshot();
});
