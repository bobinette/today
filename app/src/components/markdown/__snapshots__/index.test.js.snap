// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render markdown 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Markdown
    autoLoadReferences={false}
    className=""
    text="# Title

    A little bit of text
    "
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <strong>
          Title
        </strong>
        

        <pre>
          <code>
            A little bit of text

          </code>
        </pre>
      </div>,
      "className": "Markdown ",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": "h-1",
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <strong>
            Title
          </strong>,
          "
",
          <pre>
            <code>
              A little bit of text

            </code>
          </pre>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "h-2",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Title",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Title",
          ],
          "type": "strong",
        },
        "
",
        Object {
          "instance": null,
          "key": "h-3",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <code>
                A little bit of text

              </code>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "h-4",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "A little bit of text
",
                ],
              },
              "ref": null,
              "rendered": Array [
                "A little bit of text
",
              ],
              "type": "code",
            },
          ],
          "type": "pre",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <strong>
            Title
          </strong>
          

          <pre>
            <code>
              A little bit of text

            </code>
          </pre>
        </div>,
        "className": "Markdown ",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": "h-1",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <strong>
              Title
            </strong>,
            "
",
            <pre>
              <code>
                A little bit of text

              </code>
            </pre>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "h-2",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Title",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Title",
            ],
            "type": "strong",
          },
          "
",
          Object {
            "instance": null,
            "key": "h-3",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <code>
                  A little bit of text

                </code>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "h-4",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "A little bit of text
",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "A little bit of text
",
                ],
                "type": "code",
              },
            ],
            "type": "pre",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
